pkg/util/constants.go

```diff
@@ -22,6 +22,9 @@ const (
 
 	// FederatedResourceQuotaNameLabel is added to Work to specify associated FederatedResourceQuota's name.
 	FederatedResourceQuotaNameLabel = "federatedresourcequota.karmada.io/name"
+
+	// ManagedByWellKnownLabel is a well known label to specify the controller that manages the resource
+	ManagedByWellKnownLabel = "app.kubernetes.io/managed-by"
```

Comment:
```suggestion
	ManagedByWellKnownLabel = "app.karmada.io/managed-by"
```
I'd prefer to use the "app.`karmada`.io" instead of "app.`kubernetes`.io" as per the discussion at https://github.com/karmada-io/karmada/pull/3262#issuecomment-1489905627 and https://github.com/karmada-io/karmada/pull/3262#issuecomment-1501284793.

Reply:
I was honestly hoping to go with the [Kubernetes well-known labels](https://kubernetes.io/docs/reference/labels-annotations-taints/) but for my use-case, any identifier will work.

I will have to also rename the variable/comments to no longer suggest it's a well-known label.

Reply:
Also, if the key has karmada in it, "app.karmada.io/managed-b", perhaps the value being the actual controller (rather than `karmada`) may have been better.

With that being said, I have applied the changes as you requested.

i.e. `app.karmada.io/managed-by: karmada`

Reply:
If there is no side effect, I would love to adopt Kubernete's well-known labels, because that will be easier for users to understand their meaning.  But there is a risk that Karmada might unexpectedly overwrite the label (see https://github.com/karmada-io/karmada/pull/3262#issuecomment-1489905627).  That's the concern that drives me to go with another label.

> I will have to also rename the variable/comments to no longer suggest it's a well-known label.

Sounds good to me. The label essentially is a `reserved` label by Karmada.

> Also, if the key has karmada in it, "app.karmada.io/managed-b", perhaps the value being the actual controller (rather than karmada) may have been better.

That's definitely a good point! 
I remember the Kubefed project also [reserved a similar label](https://github.com/kubernetes-retired/kubefed/blob/d6f10d29c3785dc55abc954dec0afeddce4893ef/pkg/controller/util/managedlabel.go#L23-L27), inspire of it, maybe we can have a similar label, like:
```golang
const (
	ManagedByKarmadaLabelKey     = "karmada.io/managed"
	ManagedByKarmadaLabelValue   = "true"
	UnmanagedByKarmadaLabelValue = "false"
)
```
If we need to label a resource in the future we are still able to introduce another label(managed-by).

What do you say? @a7i @chaunceyjiang

Reply:
Sounds good to me. As far as this PR is concerned, is it correct to say that I will be using the folllowing?

```
const (
	ManagedByKarmadaLabelKey     = "karmada.io/managed"
	ManagedByKarmadaLabelValue   = "true"
)
```

Reply:
Yes, for now, we'll probably just use the `true`.

Reply:
Currently, we are developing `karmada-operator`. With `karmada-operator`, we can declaratively create multiple karmadas.

If we have multiple karmadas, can we use this label to indicate which one manages the resources?

Like:
```
karmada.io/managed: kamrada1
karmada.io/managed: kamrada2
```

Reply:
Sorry, I missed your comments.
> Currently, we are developing karmada-operator. With karmada-operator, we can declaratively create multiple karmadas.

But I guess we can have another more straightforward label for the scenario. `karmada.io/managed` only indicates a resource is managed or not.